docs/source/conf.py

```diff
@@ -13,6 +13,7 @@
 from datetime import datetime
 
 import sphinxcontrib.bibtex.plugin
+import tuple
```

Comment:
@j-emberton This breaks the docs build - we can just the generic for typing so I think can delete this line.

Reply:
I think its this line:

`signature = signature.replace("~numpy.ndarray", "ndarray")`

That breaks the docs build. If we want we can either tolerate the extra`numpy.`, or try and find a work around to get the best presentation.

Reply:
This line is definitely failing it on my machine and RTD:

``` python
➜  docs git:(262-incorrect-argument-types--docs) ✗ make html 
Running Sphinx v7.4.7

Configuration error:
There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "/Users/dorme/Library/Caches/pypoetry/virtualenvs/pyrealm-QywIOHcp-py3.11/lib/python3.11/site-packages/sphinx/config.py", line 529, in eval_config_file
    exec(code, namespace)  # NoQA: S102
    ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dorme/Research/REALM/pyrealm/docs/source/conf.py", line 16, in <module>
    import tuple
ModuleNotFoundError: No module named 'tuple'

make: *** [html] Error 2
```

If I delete that line, I get a "successful" build: no errors but 201 warnings! A lot of those are something going wrong with the `simplify_type_hints` function.

```sh
WARNING: error while formatting signature for pyrealm.constants: Handler <function simplify_type_hints at 0x105fbcd60> for event 'autodoc-process-signature' threw an exception (exception: simplify_type_hints() takes 2 positional arguments but 7 were given)
```